SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestMultipleOwnership/SpatialTestMultipleOwnership.cpp

```diff
@@ -0,0 +1,213 @@
+		AMultipleOwnershipPawn* MultipleOwnershipPawn2 =
+			GetWorld()->SpawnActor<AMultipleOwnershipPawn>(FVector(200.0f, -300.0f, 60.0f), FRotator::ZeroRotator, FActorSpawnParameters());
+
+		RegisterAutoDestroyActor(MultipleOwnershipPawn1);
+		RegisterAutoDestroyActor(MultipleOwnershipPawn2);
+
+		FinishStep();
+	});
+
+	// All workers set a reference to the MultipleOwnershipPawns in their world, to avoid code duplication.
+	AddStep(
+		TEXT("SpatialTestMultipleOwnershipAllWorkersSetReferences"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+		[this](float DeltaTime) {
+			TArray<AActor*> FoundPawns;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), AMultipleOwnershipPawn::StaticClass(), FoundPawns);
+
+			// Make sure the correct number of MultipleOwnershipPawns is visible
+			if (FoundPawns.Num() != 2)
+			{
+				return;
+			}
+
+			// Preemptively empty the Pawns array so that we don't get any issues with index-based addressing later on in the test
+			MultipleOwnershipPawns.Empty();
+
+			for (AActor* Pawn : FoundPawns)
+			{
+				AMultipleOwnershipPawn* MultipleOwnershipPawn = Cast<AMultipleOwnershipPawn>(Pawn);
+				if (MultipleOwnershipPawn != nullptr)
+				{
+					MultipleOwnershipPawns.Add(MultipleOwnershipPawn);
+				}
+			}
+			MultipleOwnershipPawns.Sort([](AMultipleOwnershipPawn& LHS, AMultipleOwnershipPawn& RHS) -> bool {
+				return LHS.GetActorLocation().Y < RHS.GetActorLocation().Y;
+			});
+			// Double checking that we have the correct number of MultipleOwnershipPawns, just to make sure that the casting above was
+			// successful and no issues arise from that
+			if (MultipleOwnershipPawns.Num() == 2)
+			{
+				FinishStep();
+			}
+		},
+		5.0f);
+
+	// Step definition for Client 1 to send a Server RPC which is used multiple times during the test. RPCs are expected to miss twice on
+	// the first pass, then once, then 0 times..
+	FSpatialFunctionalTestStepDefinition ClientSendRPCStepDefinition;
+	ClientSendRPCStepDefinition.StepName = TEXT("ClientSendRPCsTest");
+	ClientSendRPCStepDefinition.bIsNativeDefinition = true;
+	ClientSendRPCStepDefinition.NativeStartEvent.BindLambda([this]() {
+		for (AMultipleOwnershipPawn* MultipleOwnershipPawn : MultipleOwnershipPawns)
+		{
+			MultipleOwnershipPawn->ServerSendRPC();
+		}
+
+		FinishStep();
+	});
+
+	// Client 1 sends an RPC from both MultipleOwnershipPawns
+	AddStepFromDefinition(ClientSendRPCStepDefinition, FWorkerDefinition::Client(1));
+
+	// All workers check that both MultipleOwnershipPawns have received 0 RPCs, therefore that the RPCs were ignored since the
+	// MultipleOwnershipPawns had no owner.
+	AddStep(
+		TEXT("SpatialTestMultipleOwnershipAllWorkersCheckCorrectRPCs"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+		[this](float DeltaTime) {
+			if (MultipleOwnershipPawns[0]->ReceivedRPCs == 0 && MultipleOwnershipPawns[1]->ReceivedRPCs == 0)
```

Comment:
Nit: maybe requireInt(0), but not a huge deal

Reply:
Yup, I forgot to change this one, will do now

Reply:
We're not expecting pawns to start with a non-zero ReceivedRPCs, are we? I think we can `AssertEqual` in StartEvent to achieve the same behaviour, OR we could do something similar to [this](https://github.com/spatialos/UnrealGDK/pull/2893/files#diff-4098be7c84d441ed8aa736298793b5f1f985d2b0453ebd4144714b4c0cdeb89eR173-R176) to ensure we don't change from 0 for some time.